modules/ui/tools/3dmap/3d_map.js

```diff
@@ -0,0 +1,199 @@
+import { select as d3_select } from 'd3-selection';
+
+import { Map } from '../../../3drenderer/Map';
+
+import { t } from '../../../core/localizer';
+import { uiCmd } from '../../cmd';
+
+/*
+ * ui3DMap is a ui panel containing a maplibre 3D Map for visualizing buildings, roads, and areas.
+ * @param {*} context
+ * @returns
+ */
+export function ui3DMap(context) {
+  function threeDMap(selection) {
+    let wrap = d3_select(null);
+    let _isHidden = true; // start out hidden
+    let _map;
+
+    function redraw() {
+      if (_isHidden) return;
+      updateProjection();
+      featuresToGeoJSON();
+    }
+
+    function updateProjection() {
+      // Since the bounds are intended to wrap a box around a perfectly orthogonal view,
+      // for a pitched, isometric view we need to enlarge the box a bit to display more buildings.
+      let extent = context.map().extent();
```

Comment:
We should figure out the actual visible extent eventually (maybe maplibre can give it to us), but for now this is fine 👍

Reply:
Yeah, the extent gets woogity when considering the fact that we have a pitch / bearing and are elevated- let me do some digging to see if I can figure out a better way to do this